numpyro/distributions/directional.py

```diff
@@ -433,25 +433,30 @@ def sample(self, key, sample_shape=()):
         phi_state = SineBivariateVonMises._phi_marginal(
             phi_shape,
             phi_key,
-            jnp.reshape(conc, (2, batch_size)),
-            jnp.reshape(corr, (batch_size,)),
-            jnp.reshape(eig, (2, batch_size)),
-            jnp.reshape(b0, (batch_size,)),
-            jnp.reshape(eigmin, (batch_size,)),
-            jnp.reshape(phi_den, (batch_size,)),
+            jnp.broadcast_to(jnp.reshape(conc, (2, -1)), (2, batch_size)),
+            jnp.broadcast_to(jnp.reshape(corr, -1), (batch_size,)),
+            jnp.broadcast_to(jnp.reshape(eig, (2, -1)), (2, batch_size)),
+            jnp.broadcast_to(jnp.reshape(b0, -1), (batch_size,)),
+            jnp.broadcast_to(jnp.reshape(eigmin, -1), (batch_size,)),
+            jnp.broadcast_to(jnp.reshape(phi_den, -1), (batch_size,)),
```

Comment:
We already broadcast (except for corr) in phimarginal so only reshape here is required I guess.

Reply:
It seems that the current approach is not valid if each parameter has its own batch shape. I think you need to do broadcasting in the construction, and revert changes in this PR.